File: backend/ondgard-chat/src/main/java/com/ondgard/game/chat/contract/sse/SseEventType.java
package com.ondgard.game.chat.contract.sse;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum SseEventType {

  STARTED("started"),
  PROGRESS("progress"),
  COMPLETED("completed"),
  ERROR("error");

  private final String value;
}

File: frontend/src/contract/sseEventType.ts
export const SseEventType = {
  STARTED: 'started',
  PROGRESS: 'progress',
  COMPLETED: 'completed',
  ERROR: 'error',
  ADVISOR_STARTED: 'advisor_started',
  ADVISOR_THINKING: 'advisor_thinking',
  ADVISOR_COMPLETED: 'advisor_completed',
  ADVISOR_ERROR: 'advisor_error',
} as const;

export type SseEventTypeValue = (typeof SseEventType)[keyof typeof SseEventType];
